pandas/tests/io/test_gcs.py

```diff
@@ -88,16 +89,22 @@ def test_to_read_gcs(gcs_buffer, format):
     tm.assert_frame_equal(df1, df2)
 
 
-def assert_equal_zip_safe(result: bytes, expected: bytes):
+def assert_equal_zip_safe(result: bytes, expected: bytes, compression: str):
     """
-    We would like to assert these are equal, but the 10th and 11th bytes are a
-    last-modified timestamp, which in some builds is off-by-one, so we check around
-    that.
+    For zip compression, only compare the CRC-32 checksum of the file contents
+    to avoid checking the time-dependent last-modified timestamp which
+    in some CI builds is off-by-one
 
     See https://en.wikipedia.org/wiki/ZIP_(file_format)#File_headers
     """
-    assert result[:9] == expected[:9]
-    assert result[11:] == expected[11:]
+    if compression == "zip":
+        # Only compare the CRC checksum of the file contents
+        res = zipfile.ZipFile(BytesIO(result))
```

Comment:
don't you need to do these in with clauses?

Reply:
Fair point. Changed.